### PR TITLE
feat(memberships): remove content restriction handling on the homepage

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Running WooCommerce Memeberships' content restriction on the homepage is not a valid use case, but content restriction handling takes a toll on performance. This PR removes handling of content restriction on the homepage.

Code contributed by @claudiulodro 

### How to test the changes in this Pull Request:

1. Set up WP Memberships with content restriction, ensure it functions as expected 

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by claudiulodro — COMMENTED** [review]
It feels weird for me to officially approve a PR of code I mostly wrote, so I'll let someone else approve. :)

As an improvement to this, I wonder if we should/could change `is_front_page()` to `is_page() || is_archive()`? That would expand the scope of this performance enhancement to other sections, but I'm not sure if those other sections are ever supposed to actually have restricted content 🤔

**Comment by adekbadek** [comment]
> if we should/could change is_front_page() to is_page() || is_archive()? 

Hm, I'd be wary of making that broad assumptions. While I'd be on the fence about archive pages, I totally imagine use cases for content-restricting pages.

**Review by claudiulodro — APPROVED** [review]
After thinking on it, I feel OK approving this since it's already running in production on a number of sites. If you want another approval, I won't mind. :)

**Pull request merged**
